### PR TITLE
Update examples

### PR DESCRIPTION
**Pull request by pratapsingh1729**
This pull request updates some of the files in the `examples/` folder to work with the current version of Owl.  The changes are as follows:

- Changed `owl_zoo_cmd.ml` such that the `owl` shell command automatically loads the `owl-plplot` package, so that plots can be made using the scripting system.
- Changed `computation_graph_01.ml` and `computation_graph_02.ml` to work with the current functions for writing a computation graph to a dot file.
- Changed `test_log.ml` to remove some type errors.
- Changed `svm.ml` and `regression.ml` to load the `Owl_plplot` module to allow plotting.

### CODE REVIEW

File: src/zoo/owl_zoo_cmd.ml [hunk 1]
```diff
@@ -21,6 +21,7 @@ let preprocess script =
     "#use \"topfind\"\n" ^
     "#require \"owl\"\n" ^
     "#require \"owl-zoo\"\n" ^
+    "#require \"owl-plplot\"\n" ^
```
Comment (jzstark) [inline src/zoo/owl_zoo_cmd.ml]: This change breaks the design of zoo. Currently `owl-plplot` is a separate library, and requiring all the zoo script to rely on this external library is not a good choice.  If a user needs plotting function, he should require this library in a zoo script only.

File: examples/test_log.ml [hunk 2]
```diff
@@ -23,26 +22,18 @@ let generate_data () =
...
+  let y' = Mat.map (fun x -> if x > 0.5 then 1. else 0.) y' in
+  let e = Mat.((mean' (abs (y - y')))) in
   let _ = Owl_log.info "accuracy: %.4f" (1. -. e) in p
 
 let _ =
   let _ = Random.self_init () in
   let x, y = generate_data () in
   let p = test_log x y in
-  MX.pp_dsmat p;
+  Owl_pretty.print_dsnda p ;;
```
Comment (jzstark) [inline examples/test_log.ml]: No need to add ";;" in the end

File: examples/computation_graph_01.ml [hunk 3]
```diff
@@ -31,16 +39,20 @@ let visualise_03 () =
...
   let x' = adjval x |> unpack_arr |> G.arr_to_node in
   let y' = adjval y |> unpack_arr |> G.arr_to_node in
-  let s1 = G.nodes_to_dot [|x'|] in
-  let s2 = G.nodes_to_dot [|y'|] in
-  let s3 = G.nodes_to_dot [|x'; y'|] in
+  let i1 = [| unpack_arr x |> G.arr_to_node |] in
+  let s1 = G.make_graph i1 [| x' |] "graph" |> G.graph_to_dot in
+  let i2 = [| unpack_arr y |> G.arr_to_node |] in
+  let s2 = G.make_graph i2 [| y' |] "graph" |> G.graph_to_dot in
+  let s3 = G.make_graph i0 [| x' ; y' |] "graph" |> G.graph_to_dot in
```
Comment (jzstark) [inline examples/computation_graph_01.ml]: Extra space after `x'`

File: examples/computation_graph_01.ml [hunk 4]
```diff
@@ -12,16 +12,24 @@ let visualise_01 () =
...
   Sys.command "dot -Tpdf cgraph_01.dot -o cgraph_01.pdf" |> ignore
 
 
 let visualise_02 () =
   let x = G.var_elt "x" |> pack_elt in
-  let y = G.var_arr "y" |> pack_arr in
+  let y = G.var_elt "y" |> pack_elt in
   let z = (grad (f x)) y in
-  let s = [| unpack_arr z |> G.arr_to_node |] |> G.nodes_to_dot in
+  (* let s = [| unpack_arr z |> G.arr_to_node |] |> G.nodes_to_dot in *)
```
Comment (jzstark) [inline examples/computation_graph_01.ml]: Redundant comment

File: examples/computation_graph_01.ml [hunk 5]
```diff
@@ -12,16 +12,24 @@ let visualise_01 () =
   let x = G.var_elt "x" |> pack_elt in
   let y = G.var_elt "y" |> pack_elt in
   let z = f x y in
-  let s = [| unpack_elt z |> G.elt_to_node |] |> G.nodes_to_dot in
+  (* let s = [| unpack_elt z |> G.elt_to_node |] |> G.nodes_to_dot in *)
```
Comment (jzstark) [inline examples/computation_graph_01.ml]: Redundant comment

**Comment by pratapsingh1729** [comment]
Thanks very much for the comments @jzstark.  I'll fix the formatting issues and revert the changes to `test_log` and the zoo system.

**Comment by jzstark** [comment]
Cool! The code looks good and the examples work fine now. Thanks for this PR!

File: examples/computation_graph_02.ml [hunk 6]
```diff
@@ -61,16 +61,16 @@ let visualise_lstm () =
...
-
-  let s1 = adj0
+  let s1_outputs = adj0 
     |> Utils.Array.flatten
     |> Array.map (fun a -> A.unpack_arr a |> G.arr_to_node)
-    |> G.nodes_to_dot
   in
+  let s1 = G.make_graph inputs s1_outputs "mnist_loss" |> G.graph_to_dot in
   Owl_io.write_file "cgraph_04_lstm_grad.dot" s1
   (* Sys.command "dot -Tpdf -Gnslimit=1 cgraph_04_lstm_grad.dot -o cgraph_04_lstm_grad.pdf" |> ignore *)
```
Comment (ryanrhymes) [inline examples/computation_graph_02.ml]: @pratapsingh1729 fyi, there is owlviz tool which can help you to visualise the computation graph: https://github.com/owlbarn/owl/blob/master/bin/dune

**Pull request merged**